jcl/src/java.base/share/classes/jdk/internal/misc/Unsafe.java

```diff
@@ -0,0 +1,6134 @@
+		}
+	}
+
+	/**
+	 * Returns byte offset to start of class or interface
+	 * 
+	 * @param field which contains desired class or interface
+	 * @return offset to start of class or interface
+	 * 
+	 * @throws NullPointerException if field parameter is null
+	 * @throws IllegalArgumentException if field is static
+	 */
+	public long objectFieldOffset(Field field) {
+		Objects.requireNonNull(field);
+		return objectFieldOffset0(field);
+	}
+
+	/**
+	 * Returns byte offset to start of static class or interface
+	 * 
+	 * @param field which contains desired class or interface
+	 * @return offset to start of class or iterface
+	 * 
+	 * @throws NullPointerException if field parameter is null
+	 * @throws IllegalArgumentException if field is not static
+	 */
+	public long staticFieldOffset(Field field) {
+		Objects.requireNonNull(field);
+		return staticFieldOffset0(field);
+	}
+
+	/**
+	 * Returns static class or interface
+	 * 
+	 * @param field contains desired class or interface
+	 * @return static class or interface in field parameter
+	 * 
+	 * @throws NullPointerException if field parameter is null
+	 * @throws IllegalArgumentException if field is not static
+	 */
+	public Object staticFieldBase(Field field) {
+		Objects.requireNonNull(field);
+		return staticFieldBase0(field);
+	}
+
+	/**
+	 *  Determines whether class has been initialized
+	 * 
+	 * @param class to verify
+	 * @return true if method has not been initialized, false otherwise
+	 * 
+	 * @throws NullPointerException if class is null
+	 */
+	public boolean shouldBeInitialized(Class<?> c) {
+		Objects.requireNonNull(c);
+		return shouldBeInitialized0(c);
+	}
+
+	/** 
+	 * Initializes class parameter if it has not been already
+	 * 
+	 * @param c class to initialize if not already
+	 * 
+	 * @throws NullPointerException if class is null
+	 */
+	public void ensureClassInitialized(Class<?> c) {
+		Objects.requireNonNull(c);
+		ensureClassInitialized0(c);
+	}
+
+	/**
+	 * Return base offset of class array parameter
+	 * 
+	 * @param c class array
+	 * @return offset at base of array
+	 * 
+	 * @throws NullPointerException if the class parameter is null
+	 * @throws IllegalArgumentException if class is not an array
+	 */
+	public int arrayBaseOffset(Class c) {
+		Objects.requireNonNull(c);
+		return arrayBaseOffset0(c);
+	}
+
+	/**
+	 * Return index size of array in bytes
+	 * 
+	 * @param c class array
+	 * @return returns index size of array in bytes
+	 * 
+	 * @throws NullPointerException if class is null
+	 * @throws IllegalArgumentException if class is not an array
+	 */
+	public int arrayIndexScale(Class c) {
+		Objects.requireNonNull(c);
+		return arrayIndexScale0(c);
+	}
+
+	/**
+	 * @return size of address on machine in use
+	 */
+	public int addressSize() {
+		return ADDRESS_SIZE;
+	}
+
+	/**
+	 * Creates a class out of a given array of bytes with a specified ProtectionDomain.
+	 * A null ProtectionDomain will need to be resolved before the class can be used.
```

Comment:
I'm not sure what this sentence means.

Reply:
What I meant by this was the spec for `ClassLoader.defineClass` says:
"If the given ProtectionDomain is null, then a default protection domain will be assigned to the class as specified in the documentation for defineClass(String, byte[], int, int). Before the class can be used it must be resolved."

Ours sets the protectionDomain to null if it is not defined so I'm not sure if this statement still applies.

Reply:
Assigning a default ProtectionDomain, and resolving the class are two unrelated things.